### DISCUSSION TITLE
vscode + clangd add_requires之后怎么有效在main.cpp中include头文件

### DISCUSSION THREAD

**Discussion by Ramblestsad** [xmake-io/xmake-vscode#121] [Q&A] [answered]
<img width="395" alt="image" src="https://user-images.githubusercontent.com/45003009/176357860-69d898a6-e061-4170-838a-a00a92b5e391.png">
<img width="549" alt="image" src="https://user-images.githubusercontent.com/45003009/176357906-f42b9d9b-732a-43c5-9849-d5784214250b.png">
<img width="581" alt="image" src="https://user-images.githubusercontent.com/45003009/176357948-98378d00-714d-4211-8aa5-922bf77a5248.png">
<img width="916" alt="image" src="https://user-images.githubusercontent.com/45003009/176369431-88e635b3-6251-45f2-88e5-1598e683de67.png">

按照文档添加了xrepo的fmt包，我注意到有人问过类似问题但是用的是vscode的C/C++插件，我目前用的是clangd，请问如何有效include安装的fmt包？

**Comment by waruqi** [comment, marked as answer]
你这个是插件 intelligense 问题，xmake 插件本身支持自动生成 compile_commands.json 的，具体看 https://github.com/xmake-io/xmake-vscode/issues/40

至于 clangd 怎么配置使用 xmake 插件生成的 compile_commands.json ，你可以到 clangd 那边查文档。。

如果配置了还不生效，就看下 compile_commands.json 是否带上了相关的 include 路径。。如果带上了，还是不行，那就是 clangd 的 bug 

你得到 clangd/issues 那边去反馈。

**Comment by Ramblestsad** [comment]
已解决
主要是让clangd找到compile_commands.json
